pkg/controller/service/service_controller.go

```diff
@@ -240,7 +240,7 @@ func (s *ServiceController) init() error {
 // we should retry in that Duration.
 func (s *ServiceController) processServiceUpdate(cachedService *cachedService, service *v1.Service, key string) error {
 	if cachedService.state != nil {
-		if cachedService.state.UID != service.UID {
+		if cachedService.state.UID != service.UID || s.loadBalancerName(cachedService.state) != s.loadBalancerName(service) {
```

Comment:
This `cachedService` is being removed by https://github.com/kubernetes/kubernetes/pull/54569 though. There are also couple edge cases caching doesn't cover (e.g. controller restarts and service changes in between). Wonder if this orphan issue can be solved via finalizer as well.

Reply:
Crash edge cases leave orphaned objects without this change since UID change will have the same effect. Right now, some scenarios of killing cluster leaves orphan objects in Cloud environments. I saw it in Azure.
I did not find a direct description about removing `cachedService` from `processServiceUpdate()` method. But i saw some fixes applied for a case of controller destruction. Indeed, once #54569 is merged, I will need review the changes to handle controller destruction case.

Reply:
@MrHohn, can you elaborate more about changes in #54569 or advise how the prefixing of LB can be integrated with it?